apis/gateway/v1alpha1/glooedge/mutate/gateway_namespace_gateway_proxy.go

```diff
@@ -36,7 +40,37 @@ func MutateGatewayNamespaceGatewayProxy(
 		return []client.Object{original}, nil
 	}
 
-	// mutation logic goes here
+	// convert object to unstructured
+	unstructuredObj, err := resources.ToUnstructured(original)
+	if err != nil {
+		return []client.Object{original}, fmt.Errorf("failed to convert client.Object to unstructured object: %w", err)
+	}
+
+	var mutatedGateways []client.Object
+
+	// create a gateway object for each port requested
+	for _, portSpec := range parent.Spec.Ports {
+		target := unstructuredObj.DeepCopy()
+
+		spec, found, err := unstructured.NestedMap(target.Object, "spec")
+		if err != nil {
+			return mutatedGateways, fmt.Errorf("failed to retrieve spec field for gateway: %w", err)
+		}
+		if !found {
+			return mutatedGateways, fmt.Errorf("spec field not found in gateway object: %w", err)
+		}
+
+		spec["bindPort"] = portSpec.Port
+		spec["ssl"] = portSpec.SSL
+
+		if err := unstructured.SetNestedMap(target.Object, spec, "spec"); err != nil {
+			return mutatedGateways, fmt.Errorf("failed to set spec on gateway: %w", err)
+		}
+
+		target.SetName(portSpec.Name)
+
+		mutatedGateways = append(mutatedGateways, target)
+	}
```

Comment:
This logic was in the wrong function, so it was never being called. This is me moving it to the correct function.